### PR TITLE
Add Activity ID for tracing in System.Transactions

### PR DESCRIPTION
**Pull request by qizhanMS**
Add tracing for AsyncTransactionScoptTests for System.Transactions test
Fix some bugs for tracing in System.Transactions
Add ActivityID for ETW tracing for System.Transactions, use transaction ID as ActivityID

### CODE REVIEW

File: src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs [hunk 1]
```diff
@@ -203,14 +203,17 @@ internal void TransactionCreated(Transaction transaction, string type)
...
+                    TransactionCreated(transaction.TransactionTraceId.TransactionIdentifier, type);
+                else
+                    TransactionCreated(string.Empty, type);
             }
         }
 
         [Event(TRANSACTION_CREATED_EVENTID, Keywords = Keywords.TraceLtm, Level = EventLevel.Informational, Task = Tasks.Transaction, Opcode = Opcodes.Create, Message = "Transaction Created. ID is {0}, type is {1}")]
         private void TransactionCreated(string transactionIdentifier, string type)
         {
+            SetActivityId(transactionIdentifier);
```
Comment (jimcarley) [inline src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs]: >transactionIdentifier [](start = 26, length = 21)

On further reflection, the same Guid is used for all "local" transactions. The InternalTransaction object's HashCode is what differentiates the transactions. The TransactionTraceId.TransactionIdentifier string looks like "<guid>:<hashcode>". And for local transactions, the <guid> portion is always the same for the System.Transactions.dll in a given process. So using this Guid portion as the ActivityId is not useful.

File: src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs [hunk 2]
```diff
@@ -1084,5 +1198,17 @@ public class Keywords
             public const EventKeywords TraceLtm = (EventKeywords)0x0002;
             public const EventKeywords TraceDistributed = (EventKeywords)0x0004;
         }
+
+        void SetActivityId(string str)
+        {
+            Guid guid = Guid.Empty;
+
+            if (str.Length>=36)
```
Comment (dmetzgar) [inline src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs]: Is it possible to get guids both with and without dashes?

File: src/System.Transactions/tests/HelperFunctions.cs [hunk 3]
```diff
@@ -16,5 +19,52 @@ public static void PromoteTx(Transaction tx)
...
+            }
+
+            string outputString = null;
+            foreach (var actualevent in events)
+            {
+                switch (actualevent.Payload.Count)
+                {
+                    case 0:
+                        {
+                            outputString = actualevent.Message.ToString();
```
Comment (dmetzgar) [inline src/System.Transactions/tests/HelperFunctions.cs]: Message is a string, no need to call ToString() on it.

**Comment by qizhanMS** [comment]
@dotnet-bot test ci please

**Comment by qizhanMS** [comment]
@dotnet-bot test ci please

**Comment by qizhanMS** [comment]
@dotnet-bot test this please

**Comment by qizhanMS** [comment]
@dotnet-bot help

**Comment by qizhanMS** [comment]
@dotnet-bot test ci please

**Comment by jimcarley** [comment]
:shipit:

**Pull request merged**